frontend/src/metabase/ui/components/inputs/Radio/Radio.styled.tsx

```diff
@@ -14,11 +10,39 @@ export const getRadioOverrides = (): MantineThemeOverride["components"] => ({
     defaultProps: {
       size: "md",
     },
-    styles: (
-      theme: MantineTheme,
-      { labelPosition: _labelPosition }: RadioStylesParams,
-      { size = "md" },
-    ) => ({
+    variants: {
+      pill: (theme: MantineTheme) => ({
+        root: {
+          display: "inline-block",
+        },
+        inner: {
+          display: "none",
+        },
+        body: {},
+        label: {
+          fontWeight: 600,
+          display: "inline-flex",
+          alignItems: "center",
+          justifyContent: "center",
+          minWidth: "3rem",
+          height: "2rem",
+          borderRadius: theme.radius.xl,
+          backgroundColor: "var(--mb-color-brand-light)",
+          fontSize: theme.fontSizes.sm,
+          cursor: "pointer",
+          transition: "all 0.2s ease",
+          padding: 0,
+          "&:not([data-checked])": {
+            color: theme.fn.themeColor("brand"),
+          },
+          "[data-checked] &": {
+            backgroundColor: theme.fn.themeColor("brand"),
+            color: "white",
+          },
+        },
```

Comment:
this new  radio variant looks good to me. Tagging @kdoh for design review

![Screen Shot 2024-10-02 at 4 37 22 PM](https://github.com/user-attachments/assets/0b525580-d888-408a-b815-0f81060593db)

Reply:
see also: https://mantine.dev/core/chip/